### PR TITLE
HHH-11761 - Extend AbstractEntityPersister support for rowId for delete statements

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-11761

### CODE REVIEW

**Comment by vladmihalcea** [comment]
This PR is due for 6.0 only.

**Comment by sebersole** [comment]
@vladmihalcea you say this is for 6.0 and added that label.  But the PR itself targets master??

**Comment by vladmihalcea** [comment]
@sebersole I developed it on the master branch, but,  at the time, it was decided that all new features should be postponed for the 6th version. At the time, it was not possible to test it on 6, so it was left as-is.

If you want, I could rebase it.

**Comment by sebersole** [comment]
@vladmihalcea If you rebase it and @Naros agrees I can apply it

**Comment by vladmihalcea** [comment]
@sebersole I rebased the branch, resolved the conflicts and pushed it.

@Naros If you can take a look at it, that would be great. Thanks.

File: hibernate-core/src/test/java/org/hibernate/test/rowid/RowIdTest.java [hunk 1]
```diff
@@ -42,50 +56,58 @@ public void afterSessionFactoryBuilt() {
...
-		s.clear();
-		
-		t = s.beginTransaction();
-		p = (Point) s.get(Point.class, p);
-		p.setDescription("new new new desc");
-		t.commit();
-		s.close();
+	public void testUpdateByRowId() {
+		doInHibernate( this::sessionFactory, s -> {
+			Point p = new Point( new BigDecimal(1.0), new BigDecimal(1.0) );
```
Comment (kortov) [inline hibernate-core/src/test/java/org/hibernate/test/rowid/RowIdTest.java]: minor : Idea higlights it as "Unpredictable 'new BigDecimal()' call"
image

```suggestion
			Point p = new Point( new BigDecimal("1.0"), new BigDecimal("1.0") );
```

File: hibernate-core/src/test/java/org/hibernate/test/rowid/RowIdTest.java [hunk 2]
```diff
@@ -42,50 +56,58 @@ public void afterSessionFactoryBuilt() {
...
+		doInHibernate( this::sessionFactory, s -> {
+			Point p = s.get(Point.class, _p);
+			p.setDescription("new new new desc");
+		} );
 	}
 
+	@Test
+	public void testDeleteByRowId() {
+		Point _p = doInHibernate( this::sessionFactory, s -> {
+			Point p = new Point( new BigDecimal(1.0), new BigDecimal(1.0) );
```
Comment (kortov) [inline hibernate-core/src/test/java/org/hibernate/test/rowid/RowIdTest.java]: minor: the same as above

```suggestion
			Point p = new Point( new BigDecimal("1.0"), new BigDecimal("1.0") );
```

File: hibernate-core/src/test/java/org/hibernate/test/rowid/RowIdTest.java (hunk 1 above)
Comment (vladmihalcea) [reply hibernate-core/src/test/java/org/hibernate/test/rowid/RowIdTest.java]: I think that `BigDecimal.valueOf` might be a good alternative.

Comment (kortov) [reply hibernate-core/src/test/java/org/hibernate/test/rowid/RowIdTest.java]: Yep, I didn't know about that

```suggestion
			Point p = new Point( BigDecimal.valueOf(1.0), BigDecimal.valueOf(1.0) );
```

**Comment by sebersole** [comment]
Hm, not sure why I asked Chris to check.  I think I just saw so many envers-related requests that fell into this "6.0?" bucket.

Anyway... Maybe I am just missing it, but I do not see any tests that ROWID is actually used.

**Pull request closed without merging**